configs/components/facter.rb

```diff
@@ -226,6 +224,23 @@
     ["#{make} -j$(shell expr $(shell #{platform[:num_cores]}) + 1) install"]
   end
 
+  if platform.is_osx?
+    ldd = "otool -L"
+  else
+    ldd = "ldd"
+  end
+
+  unless platform.is_windows? || platform.is_cross_compiled_linux? || platform.architecture == 'sparc'
+    pkg.check do
+      [
+        # Ensure we're linking against our libstdc++ and libgcc_s or statically linking
+        "#{ldd} lib/libfacter.so",
+        "[ ! $$(#{ldd} lib/libfacter.so | grep -c libstdc++) -ge 1 ] || #{ldd} lib/libfacter.so | grep libstdc++ | grep puppet || #{ldd} lib/libfacter.so | grep libstdc++ | grep pl-build-tools",
+        "[ ! $$(#{ldd} lib/libfacter.so | grep -c libgcc_s) -ge 1 ] || #{ldd} lib/libfacter.so | grep libgcc_s | grep puppet || #{ldd} lib/libfacter.so | grep libgcc_s | grep pl-build-tools"
```

Comment:
These initial tests (`[]`) are kind of janky. Why are we inverting the test conditional in there? Wouldn't this be the same:

```bash
mckern@weth puppet $ [ ! $(ldd lib/libfacter.so | grep -c libstdc++) -ge 1 ] && echo "true" || echo "false"
true
mckern@weth puppet $ [ $(ldd lib/libfacter.so | grep -c libstdc++) -eq 0 ] && echo "true" || echo "false"
true
mckern@weth puppet $
```